File: DotNet/Submission/Application/Models/PatientSubmissionModel.cs
using Hl7.Fhir.Model;
using LantanaGroup.Link.Shared.Application.Converters;
using LantanaGroup.Link.Shared.Domain.Entities;
using System.Text.Json.Serialization;

namespace LantanaGroup.Link.Submission.Application.Models
{
    public class PatientSubmissionModel : BaseEntityExtended
    {
        public string FacilityId { get; set; } = string.Empty;
        public string PatientId { get; set; } = string.Empty;
        public string ReportScheduleId { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        [JsonConverter(typeof(FhirResourceConverter<Bundle>))]
        public Bundle PatientResources { get; set; }
        [JsonConverter(typeof(FhirResourceConverter<Bundle>))]
        public Bundle OtherResources { get; set; }
    }
}

File: Java/validation/src/main/java/com/lantanagroup/link/validation/entities/PatientSubmissionModel.java
package com.lantanagroup.link.validation.entities;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.lantanagroup.link.shared.serdes.FhirIdDeserializer;
import lombok.Getter;
import lombok.Setter;
import org.hl7.fhir.r4.model.Bundle;

import java.util.Date;

@Getter
@Setter
public class PatientSubmissionModel {
    private String facilityId;

    @JsonDeserialize(using = FhirIdDeserializer.class)
    private String patientId;

    private String reportScheduleId;
    private Date startDate;
    private Date endDate;
    private Bundle bundle;
}
